test/cutting/test_cutting_roundtrip.py

```diff
@@ -179,28 +179,56 @@ def test_cutting_exact_reconstruction(example_circuit):
     assert np.allclose(exact_expvals, simulated_expvals, atol=1e-8)
 
 
-def test_sampler_with_identity_subobservable(example_circuit):
-    """This test ensures that the sampler does not throw an error if you pass it a subcircuit with no observable measurements.
+@pytest.mark.parametrize(
+    "sampler,is_exact_sampler", [(Sampler(), False), (ExactSampler(), True)]
+)
+def test_sampler_with_identity_subobservable(sampler, is_exact_sampler):
+    """This test ensures that the sampler works for a subcircuit with no observable measurements.
 
-    Tests temporary workaround to Issue #422.
+    Specifically, that
 
-    This test passes if no exceptions are raised.
+    - ``Sampler`` does not blow up (Issue #422); and
+    - ``ExactSampler`` returns correct results
 
+    This is related to https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/issues/422.
     """
+    # Create a circuit to cut
+    qc = QuantumCircuit(3)
+    append_random_unitary(qc, [0, 1])
+    append_random_unitary(qc, [2])
+    qc.rxx(np.pi / 3, 1, 2)
+    append_random_unitary(qc, [0, 1])
+    append_random_unitary(qc, [2])
 
```

Comment:
What motivates the choice of this particular circuit here?

Reply:
It's sufficiently simple, but not trivial in any way, either.  It has two qubits involved in cutting, and a third qubit that gets entangled both before and after the cut but is otherwise "along for the ride."

Reply:
It's also a special case of a circuit generated by the `example_circuit` fixture, which was used before.  But we don't need to spend time testing against every single one of them.  Just one should be sufficient, and is enough to fail without me fixing the line with `_get_pauli_indices`.

Reply:
So is there something significant that is lost if we do away with appending random unitaries before and after adding the rxx?

Reply:
If we got rid of the random unitaries before the cut gate, we'd only be testing that cutting works under the assumption that |0> is the input state, so there could be bugs that fall through.

Likewise, if we got rid of the random unitaries after the cut gate, then we'd only be ensuring that the post-cut state has the right amplitude in the Z basis, again, a much more limited test compared to the current, which should only reliably work if the post-cut state is correct in all bases.

Reply:
Got it, thank you so patiently clarifying this.